frontend/src/main/scala/bloop/engine/Build.scala

```diff
@@ -32,8 +34,11 @@ final case class Build private (
     val files = projects.iterator.map(p => p.origin.toAttributedPath).toSet
     val newFiles = BuildLoader.readConfigurationFilesInBase(origin, logger).toSet
 
+    val loadedSettings = WorkspaceSettings.fromFile(origin, logger)
+    val sameSettings = loadedSettings == settings
+
     // This is the fast path to short circuit quickly if they are the same
-    if (newFiles == files) Task.now(Build.ReturnPreviousState)
+    if (newFiles == files && sameSettings) Task.now(Build.ReturnPreviousState)
```

Comment:
Can we instead just check that the semanticdb version is the same in both? This way, when we add new fields to settings that do not require transformations in the project, we will not be reloading needlessly.